Source/Core/AudioCommon/WASAPIStream.h

```diff
@@ -35,18 +38,21 @@ class WASAPIStream final : public SoundStream
 
   static bool isValid();
   static std::vector<std::string> GetAvailableDevices();
-  static IMMDevice* GetDeviceByName(std::string name);
+  static Microsoft::WRL::ComPtr<IMMDevice> GetDeviceByName(std::string name);
 
 private:
   u32 m_frames_in_buffer = 0;
   std::atomic<bool> m_running = false;
   std::atomic<bool> m_stopped = false;
   std::thread m_thread;
 
-  IAudioClient* m_audio_client = nullptr;
-  IAudioRenderClient* m_audio_renderer = nullptr;
-  IMMDeviceEnumerator* m_enumerator = nullptr;
-  HANDLE m_need_data_event = nullptr;
+  // This MUST be located before ComPtr's underneath!
```

Comment:
Could you please add more detail to this comment? Such as "must be located before...to ensure CoInitialize is called/COM stuff works/for reasons" or whatever makes most sense to describe the _why_.